File: Angular-Project/Pizza-Order-App/src/app/models/Pizza.ts
export class Pizza {
	private _id: number;

	private _name: string;

	private _description: string;

	private _imageUrl: string;

	private _price: number;

	constructor(
		id: number,
		name: string,
		description: string,
		imageUrl: string,
		price: number
	) {
		this._id = id;
		this._name = name;
		this._description = description;
		this._imageUrl = imageUrl;
		this._price = price;
	}

	get id(): number {
		return this._id;
	}

	get name(): string {
		return this._name;
	}

	get description(): string {
		return this._description;
	}

	get imageUrl(): string {
		return this._imageUrl;
	}

	get price(): number {
		return this._price;
	}
}

File: Spring-Project/src/main/java/hu/dominikvaradi/entities/Pizza.java
package hu.dominikvaradi.entities;

import org.hibernate.Hibernate;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "pizza")
public class Pizza {
    /**
     * TODO kép tárolás
     */


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "name", nullable = false, length = 50)
    private String name;

    @Lob
    @Column(name = "description")
    private String description;

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Pizza pizza = (Pizza) o;
        return id != null && Objects.equals(id, pizza.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}